Upgrade cms for new i18n backend

tractorcow opened the pull request:
Required for https://github.com/silverstripe/silverstripe-framework/pull/6558

mikenz commented on .editorconfig:
```diff
@@ -6,7 +6,7 @@
 end_of_line = lf
 indent_size = 4
-indent_style = tab
+indent_style = space
```
Yay! But was this really supposed to be in this PR?

tractorcow replied:
We should be using psr2 now; I think some upgrades were missing from cms module though. I should run a formatter over it.

tractorcow replied:
Oh look, we never merged it. =( https://github.com/silverstripe/silverstripe-cms/pull/1688

tractorcow replied:
Rebuilt at https://github.com/silverstripe/silverstripe-cms/pull/1727. I'll revert from this PR.

tractorcow commented:
I've reverted any presumptuous psr2 reformatting on SiteTree.php; I'll wait until we merge the main psr2 PR instead :D

tractorcow commented:
Ok, I've updated the tests and fixed some german localisations for plurals. Should be passing now. :D

note: CMS pr can be merged ahead of the framework, since it only fixes problems, it doesn't RELY on the new api for it to work.

tractorcow commented:
This is ready to merge.
